src/lib/interface/robotInfo.ts

```diff
@@ -7,7 +7,7 @@ class RobotInfo {
   constructor(givenROS) {
     this.myROS = givenROS;
 
-    if (this.myROS != null) {
+    if (this.myROS !== null) {
```

Comment:
Unless I'm misunderstanding this code, `!= null` is actually a valid check here, because it covers both `null` and `undefined`. Generally `!==` and `===` are what you should use, except in the case of `null` if you don't care about whether something is `null` or `undefined`, just that it doesn't have a value.

So, any `!== null` and `=== null` should be replaced with their corresponding single equals varieties when we don't care about whether the value is `null` or `undefined`, just that it doesn't exist. I won't comment on any other times this shows up, so it's probably a good idea to run a full-code search.

Reply:
Nice catch. This robot info needs to actually be refactored entirely. One of the main reasons I chose Pinia is that the useRosStore and the roslib object state itself is always defined, aka the check is not actually necessary.

Reply:
Noted for the future.